CMakeLists.txt

```diff
@@ -1,18 +1,28 @@
-cmake_minimum_required(VERSION 3.16)
-# https://semver.org/
+cmake_minimum_required(VERSION 3.24)
+
 project(farm_ng_core VERSION 0.2.0)
 
-set(farm_ng_cmake_DIR ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
-include(${farm_ng_cmake_DIR}/farm_ng_cmakeConfig.cmake)
-farm_ng_module(farm_ng_core)
+# Custom farm-ng macros
+include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/farm_ng_cmakeConfig.cmake)
 
-message("CMAKE_CXX_FLAGS" ${CMAKE_CXX_FLAGS})
+# Support local development package override
+include(farm_ng_dev_packages)
+# Support ccache
+include(farm_ng_ccache)
 
-farm_ng_enable_testing()
+find_package(fmt REQUIRED)
+find_package(GTest REQUIRED) 
+find_package(Eigen3 3.4.0 REQUIRED)
+find_package(Protobuf REQUIRED)
+find_package(gRPC REQUIRED)
+find_package(CLI11 REQUIRED)
+find_package(Boost REQUIRED COMPONENTS system)
+find_package(Sophus 2.0.0 REQUIRED)
 
-set(CMAKE_CXX_CLANG_TIDY "")
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
 
-find_package(fmt REQUIRED)
+enable_testing()
 
-add_subdirectory(protos)
 add_subdirectory(cpp)
+add_subdirectory(protos)
```

Comment:
I think it may be wise to generate protobuf files first so we can express dependencies on the protos from the cpp files. If I recall correctly, CMake won't recognize the dependency if we haven't generated it yet.

Reply:
Interesting. How would this issue manifest / is there a way to test that?